spec/controllers/coronavirus_form/address_lookup_controller_spec.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe CoronavirusForm::AddressLookupController, type: :controller do
+  include_examples "redirections"
+  include_examples "session expiry"
+
+  let(:current_template) { "coronavirus_form/address_lookup" }
+  let(:session_key) { :support_address }
+  let(:next_page) { support_address_path }
+
+  before do
+    VCR.turn_on!
+  end
+
+  after do
+    VCR.turn_off!
+  end
+
+  describe "GET show" do
+    it "renders the postcode form" do
+      session[:live_in_england] = I18n.t("coronavirus_form.questions.live_in_england.options.option_yes.label")
+
+      get :show
+      expect(response).to render_template(current_template)
+    end
+  end
+
+  describe "POST submit" do
+    context "no postcode" do
+      it "renders an error message if the postcode is not given" do
+        post :submit, params: { "postcode" => "" }
+
+        expect(response).to have_http_status(:unprocessable_entity)
+        expect(response).to render_template(current_template)
+      end
+    end
+
+    context "invalid postcode" do
+      it "renders an error message if the postcode is invalid" do
+        VCR.use_cassette "/address/400_response" do
+          post :submit, params: { "postcode" => "AA1A1AA" }
+
+          expect(response).to have_http_status(:unprocessable_entity)
+          expect(response).to render_template(current_template)
+        end
+      end
+    end
+
+    context "valid postcode" do
+      it "renders the API results given a valid postcode" do
+        VCR.use_cassette "/address/200_individual_building" do
+          post :submit, params: { "postcode" => "SW1A2AA" }
+
+          expect(response).to have_http_status(:ok)
+          expect(response).to render_template(current_template)
+        end
+      end
+    end
```

Comment:
Would be good to have a test for what happens if they have already viewed the check_your_answers page here.

Reply:
Done.